### DISCUSSION TITLE
Improve UI and customisability

### DISCUSSION THREAD

**Discussion by scratchmex** [authelia/authelia#7469] [Ideas]
I really like the philosophy of authelia, it's simplicity, but I don't like it's design. The UI is not shiny for me.

I would like to be able to change the background overall "theme" like the background of the login page 

I am open to make a code change proposal of the UI. I would like to discuss how much customisability are the maintainers willing to make

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It's great to hear that you appreciate Authelia's philosophy and simplicity! Customizing the UI, including the background and overall theme, is a common request among users. The maintainers of Authelia are open to contributions and discussions regarding UI customizability.

To start, you can open a discussion on GitHub to propose your ideas and get feedback from the maintainers and the community. This will help you understand how much customizability the maintainers are willing to incorporate and any guidelines you should follow.

Here are some steps you can take:

1. **Open a Discussion on GitHub**: Start a discussion to propose your ideas for UI customization. Explain what changes you would like to make and why you think they would be beneficial.

2. **Follow Contribution Guidelines**: Make sure to read and follow the contribution guidelines provided by Authelia. This will help ensure that your proposal aligns with the project's standards.

3. **Design and Implementation**: Once you have feedback and approval from the maintainers, you can start working on your code changes. Be sure to keep the maintainers updated on your progress and seek feedback as needed.

4. **Submit a Pull Request**: After implementing your changes, submit a pull request for review. The maintainers will review your code and provide feedback or merge it if it meets the project's requirements.

You can start the discussion on GitHub here: [Authelia GitHub Discussions](https://github.com/authelia/authelia/discussions)

For more information on contributing, you can refer to the following documentation:
- [Contributing to Authelia](https://www.authelia.com/contributing/prologue/introduction)
- [Development Guidelines](https://www.authelia.com/contributing/development/introduction)

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Explain what you mean by changing the background in more detail? I'm not sure we'd ever give end users the ability to set background images, but I think if properly validated we'd be happy to allow them to set colors. For clarity allowing users to upload binary content is a practice which is likely to have a significant security peril attached. Even if the users themselves are trustworthy it often allows additional attack vectors for non-users.

The ability of admins on the other hand is not that big a deal, assuming the requirements for adding the image are exact and rigid , and it's thoroughly tested on multiple device layouts (mobile etc), then we would likely be able to support it. If we make the requirements wishy washy and we then have to elaborate on the requirements regularly we'd be unwilling to support this. 

I had in mind allowing admins to define custom themes and give users the choice in the frontend to pick between them. Each theme would theoretically have a light and dark variant, and we'd have a mode switcher remembered in localStorage. If the mode switcher had not been interacted with the theme would default to the media query preference.

**Comment by scratchmex** [comment]
something like this

- globally, either change the background color or give a base64 encoded image to be displayed
  - this is useful to distinguish uniquely your authelia instance
  - in case of the background color, it can be part of a theme which also affects the buttons like the "Sign In" here
- per user, have an base64 encoded image to be displayed as icon
  - this is useful in setups like with lldap where you can upload your profile picture

![image](https://github.com/user-attachments/assets/6d46c947-44cd-4f1b-9601-09eff50c70ca)

**Comment by JervisTetch** [comment]
I think a great solution would be overrideable templates ([like Gitea does](https://docs.gitea.com/administration/customizing-gitea#customizing-gitea-pages-and-resources)), this way users could easily utilize Authelia as a fully customizable user portal.

In my use cases I often have some sort of simple user control panel where users can change their account settings and I just need Authelia in front of it, a possibility of having everything (frontend-wise) in one place (Authelia) for smaller setups sounds appealing to me.

**Reply by james-d-elliott** [reply]
While this is a nice concept it's not something we'll invest time into since we're heavily using react. The likely implementation is two fold.

First that we'll allow creating custom themes based on a subset of the MUI themes API. Effectively this would come in the form of creating custom color palletes in YAML and then linking those to an overarching theme name (one for light mode and one for dark mode). 

Secondly adding the ability to provide a custom.css file.

**Reply by JervisTetch** [reply]
I see, so it's basically the necessary pre-compilation introduced by React that is making template overriding overly complex?

I'm not deep enough into the codebase yet, my wishful thinking was that we could just implement a simple override mechanism for all relevant files in `web/src` to make templating extremely flexible.

**Reply by james-d-elliott** [reply]
We only use minimal templates for very specific assets that need it, the frontend is a typescript react app. We have no intention of changing this.

**Comment by prakash-halan** [comment]
@james-d-elliott 
Can you please tell how to change the label text here in login page for "Sign In" in the top and the label of the button "SIGN IN"
When I try using locales en portal.json , it changes both the text, Case sensitive doesn't work.  Its really difficult to customise 
In top I need to add the application name  something like "My App - Sign In" kind of title

**Reply by lokomass** [reply]
Same problem for me…

**Reply by james-d-elliott** [reply]
@prakash-halan I'm not sure I'm following what you mean, would you mind showing a screenshot?

**Reply by prakash-halan** [reply]
<img width="472" height="340" alt="Screenshot 2025-10-28 at 10 28 11 AM" src="https://github.com/user-attachments/assets/843b976b-3e57-4e90-8290-52823523a3e7" />

@james-d-elliott , Please check the image attached. We have "Sign in" in the top and "SIGN IN" as button label too. We can't change them individually . If I add a locale en for "Sign in" it changes for "SIGN IN" also.

**Reply by james-d-elliott** [reply]
Yeah that's generally expected, generally for each individual string we'd only include one instance of it. What's the purpose behind changing one and not the other?

**Reply by prakash-halan** [reply]
Purpose is to give  "product name - Sign In" to make the UI more appealing. There is no place to put the product name or some title to make the UI more attractive and appealing . Mostly the UX team wants us to show the product name in the login page 
I was able to make that changes using Nginx sub_filter for now.

**Reply by lokomass** [reply]
> Yeah that's generally expected, generally for each individual string we'd only include one instance of it. What's the purpose behind changing one and not the other?

In the portal.json, you can separate two items :

```
{
  ...
  "Sign in title": "Sign in",
  "Sign in button": "Sign in",
  ...
}
```

or 

```
{
  ...
  "Sign in": {
    "title": "Sign in",
    "button": "Sign in"
  }
  ...
}
```

**Reply by james-d-elliott** [reply]
> Purpose is to give "product name - Sign In" to make the UI more appealing. There is no place to put the product name or some title to make the UI more attractive and appealing . Mostly the UX team wants us to show the product name in the login page
I was able to make that changes using Nginx sub_filter for now.

We can look at something for this in 4.40 or 4.41.

> In the portal.json, you can separate two items

Neither of these options are actually viable as they break fundamental i18n principles, while the translations can be used to do some level of customization the principal goal is accessibility.

**Reply by prakash-halan** [reply]
Thanks @james-d-elliott . It will be helpful if we can do something for this. 
Just a suggestion , the bottom button label "Sign In" can be renamed as "Submit" or something similar to that.

**Reply by james-d-elliott** [reply]
What about a prefix to the `sign in` at the top which is by default an empty string?

**Reply by prakash-halan** [reply]
Yes, anything is fine as long as it’s customizable.

**Reply by lokomass** [reply]
But then, we won’t be able to use “Authelia” as the title and “Sig in” on the button?

**Comment by valentimarco** [comment]
news?
